python/ray/dag/compiled_dag_node.py

```diff
@@ -1542,6 +1562,18 @@ def run(self):
         monitor.start()
         return monitor
 
+    def raise_if_too_many_inflight_requests(self):
+        num_in_flight_requests = self._execution_index - self._max_execution_index
+        if num_in_flight_requests > self._max_buffered_inputs:
+            raise ray.exceptions.RayChannelBufferAtMaxCapacity(
```

Comment:
What happens if the condition `self.num_in_flight_requests > self._max_buffered_inputs` is met? Will it cause a deadlock, or is it possible for the execution to still finish but slower?

If it is still possible to finish, maybe we should consider using a warning message instead of raising an exception. Consider the case where a job has already been running for a long time but finally fails because it executes too frequently in the last few steps.

Reply:
Then it is the same semantic as before. It can timeout depending on how many channels are available. (this is the problem we want to fix. Basically problem 1 in this doc https://docs.google.com/document/d/1Ka_HFwPBNIY1u3kuroHOSZMEQ8AgwpYciZ4n08HJ0Xc/edit#heading=h.8j5v3i9ykgsy). 

This raises when .execute() is called, so actually we have no problem you mentioned.